### DISCUSSION TITLE
Markdown files open in browser

### DISCUSSION THREAD

**Discussion by nthieu99** [DirectoryLister/DirectoryLister#738] [Help & Support] [answered]
Hello,

When I click on the Markdown files (README.md), it open in the browser (e.g. domain.com/README.md) instead of downloading the file. How can I make it download the file when clicking on?

Thanks for your answers.

**Comment by PHLAK** [comment, marked as answer]
Directory Lister does not handle the way files are opened directly. This is handled in combination by the web server and the user's web browser.

First and foremost ensure your server is configured to [return the correct MIME type](https://developer.mozilla.org/en-US/docs/Learn/Server-side/Configuring_server_MIME_types) of files.

  - For Apache this is accomplished with the [`AddType` directive](https://httpd.apache.org/docs/current/mod/mod_mime.html#addtype) (part of `mod_mime`)
  - For NGINX this is handled by the [`types` configuration option](http://nginx.org/en/docs/http/ngx_http_core_module.html#types) (usually found in the [`mime.types`](https://www.nginx.com/resources/wiki/start/topics/examples/full/#mime-types) file)

Additionally, the user's browser may be configured to handle all files of a given type in a specific fashion (i.e. open or download). Reference your browsers support page (e.g. [Firefox](https://support.mozilla.org/en-US/kb/change-firefox-behavior-when-open-file), [Chrome](https://support.google.com/chrome/answer/95759)) for instructions on configuring this.

**Comment by nthieu99** [comment]
I had achieved this by adding the the following code into .htaccess (Apache server).

`<Files`` *.md>
    ForceType application/octet-stream
    Header set Content-Disposition attachment
</Files>`

Is this a good way to achieve this?

**Reply by PHLAK** [reply]
I haven't really looked into Apache configurations in a while but this seems okay to me.
